internal/reflect/handler.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package reflect
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/dynamic"
+	intdesc "github.com/uber/prototool/internal/desc"
+	"github.com/uber/prototool/internal/x/extract"
+	"go.uber.org/zap"
+)
+
+type handler struct {
+	logger *zap.Logger
+
+	getter extract.Getter
+}
+
+func newHandler(options ...HandlerOption) *handler {
+	handler := &handler{
+		logger: zap.NewNop(),
+	}
+	for _, option := range options {
+		option(handler)
+	}
+	// TODO(pedge): composition
```

Comment:
can we drop this todo?

Reply:
We can, but I still want to do it...so... :-)

Reply:
haha, then let's do it?!

Reply:
I need to do this in like three places...perhaps this should be `// TODO: composition?` like as in a question lol

Reply:
lol sounds good, let's merge it